core/src/test/scala/com/databricks/labs/remorph/parsers/common/FunctionBuilderSpec.scala

```diff
@@ -283,39 +282,71 @@ class FunctionBuilderSpec extends AnyFlatSpec with Matchers with TableDrivenProp
 
   "buildFunction" should "remove quotes and brackets from function names" in {
     // Test function name with less than 2 characters
-    val result1 = FunctionBuilder.buildFunction("a", Seq())
+    val result1 = FunctionBuilder.buildFunction("a", List.empty, TSql)
     result1 match {
-      case f: UnresolvedFunction => f.function_name shouldBe "a"
+      case f: ir.UnresolvedFunction => f.function_name shouldBe "a"
       case _ => fail("Unexpected function type")
     }
 
     // Test function name with matching quotes
-    val result2 = FunctionBuilder.buildFunction("'quoted'", Seq())
+    val result2 = FunctionBuilder.buildFunction("'quoted'", List.empty, TSql)
     result2 match {
-      case f: UnresolvedFunction => f.function_name shouldBe "quoted"
+      case f: ir.UnresolvedFunction => f.function_name shouldBe "quoted"
       case _ => fail("Unexpected function type")
     }
 
     // Test function name with matching brackets
-    val result3 = FunctionBuilder.buildFunction("[bracketed]", Seq())
+    val result3 = FunctionBuilder.buildFunction("[bracketed]", List.empty, TSql)
     result3 match {
-      case f: UnresolvedFunction => f.function_name shouldBe "bracketed"
+      case f: ir.UnresolvedFunction => f.function_name shouldBe "bracketed"
       case _ => fail("Unexpected function type")
     }
 
     // Test function name with matching backslashes
-    val result4 = FunctionBuilder.buildFunction("\\backslashed\\", Seq())
+    val result4 = FunctionBuilder.buildFunction("\\backslashed\\", List.empty, TSql)
     result4 match {
-      case f: UnresolvedFunction => f.function_name shouldBe "backslashed"
+      case f: ir.UnresolvedFunction => f.function_name shouldBe "backslashed"
       case _ => fail("Unexpected function type")
     }
 
     // Test function name with non-matching quotes
-    val result5 = FunctionBuilder.buildFunction("'nonmatching", Seq())
+    val result5 = FunctionBuilder.buildFunction("'nonmatching", List.empty, TSql)
     result5 match {
-      case f: UnresolvedFunction => f.function_name shouldBe "'nonmatching"
+      case f: ir.UnresolvedFunction => f.function_name shouldBe "'nonmatching"
       case _ => fail("Unexpected function type")
     }
   }
 
+  "buildFunction" should "Apply known TSQL conversion strategies" in {
```

Comment:
as a side note, we could have a single block for tests around `buildFunction` like in:
```scala
"buildFunction" should {
  "Apply known TSQL conversion strategies" in {
    // ...
  }
  // etc.
}
```